cdap-docs/developers-manual/source/building-blocks/applications.rst

```diff
@@ -48,9 +48,77 @@ Other components are defined using user-written classes that implement correspon
 interfaces and are referenced by passing an object, in addition to being assigned a unique
 name.
 
-Names used for streams and datasets need to be unique across the CDAP instance, while
+Names used for streams and datasets need to be unique across the CDAP namespace, while
 names used for programs and services need to be unique only to the application.
 
+Applications can also use a Configuration class to receive a configuration during deployment time of the Application.
+For example, this can be used to specify the stream, dataset to create/read from at application deployment time
```

Comment:
For example, this can be used to specify that |---| at application deployment time |---| a stream be created or a dataset
be read from, rather than hard-coding them in the...